src/librustc_typeck/diagnostics.rs

```diff
@@ -3426,6 +3426,30 @@ parameters. You can read more about it in the API documentation:
 https://doc.rust-lang.org/std/marker/struct.PhantomData.html
 "##,
 
+E0393: r##"
+A type parameter which needs to be explicity specified because its
```

Comment:
This is confusing. How about "A type parameter which references `Self` in its default value was not specified.".